features/step_definitions/api_request_steps.rb

```diff
@@ -67,48 +84,32 @@
   @proceeding_type_data = { "proceeding_types": [{ ccms_code: CFEConstants::ASYLUM_PROCEEDING_TYPE_CCMS_CODE, client_involvement_type: "A" }] }
 end
 
-Given("I create an assessment with the following details:") do |table|
-  data = table.rows_hash
-
-  if data.key?("proceeding_types")
-    data["proceeding_types"] = { 'ccms_codes': data["proceeding_types"].split(";") }
-  end
-
-  @assessment_data = data.symbolize_keys
-end
-
-Given("I add the following applicant details for the current assessment:") do |table|
-  @applicant_data.merge! cast_values(table.rows_hash)
+Given("I have a dependant aged {int}") do |age|
+  date_of_birth = Date.parse(@assessment_data[:submission_date]) - age.years
+  @dependant_data[:dependants] << { date_of_birth: date_of_birth.to_s, in_full_time_education: true, relationship: "child_relative" }
 end
 
-Given("I add the following dependent details for the current assessment:") do |table|
-  data = table.hashes.map do |h|
-    amount = h["income_amount"]
-    frequency = h["income_frequency"]
-    if amount.present? && frequency.present?
-      h = cast_values(h)
-      h["income"] = { "amount" => amount.to_f, "frequency" => frequency }
-      h.except("income_amount", "income_frequency")
-    else
-      cast_values(h)
-    end
-  end
-  deps = data.map do |d|
-    d.symbolize_keys.slice(:date_of_birth, :in_full_time_education,
-                           :relationship, :income, :monthly_income, :income_frequency, :assets_value)
-  end
-  @dependant_data = {
-    "dependants": deps,
+Given("I have a dependant aged {int} with monthly income of {int}") do |age, income|
+  date_of_birth = Date.parse(@assessment_data[:submission_date]) - age.years
+  @dependant_data[:dependants] << {
+    date_of_birth: date_of_birth.to_s,
+    in_full_time_education: true,
+    monthly_income: income,
+    relationship: "child_relative",
   }
 end
 
-Given("I add the following other_income details for {string} in the current assessment:") do |string, table|
-  @other_incomes_data = { "other_incomes": [{ "source": string, "payments": table.hashes.map { cast_values(_1) } }] }
-end
-
-Given("I add the following housing benefit details for the applicant:") do |table|
-  @benefits_data = { state_benefits: [{ "name": "housing_benefit",
-                                        "payments": table.hashes.map { cast_values(_1) } }] }
+Given("I have a dependant aged {int} with {string} income of {int}") do |age, frequency, income|
```

Comment:
this is quite similar to the method above:
`I have a dependant aged {int} with monthly income of {int}`
Would it be better to use this method, make 'monthly' an acceptable input of this method and delete the monthly specific helper above.

Reply:
There are 2 ways of specifying income - IIRC the 'monthly' one uses the old method, this uses the more flexible version